loaddb.py

```diff
@@ -1255,6 +1276,35 @@ def create_master_reader():
                 [ Logfile(x, blacklist) for x in LOGS ])
   return MasterXlogReader(processors)
 
+def load_args():
+  parser = argparse.ArgumentParser(
+    formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+  parser.add_argument("--db-retry-connect", action='store_true',
+    help="Retry connecting to the database forever.")
+  parser.add_argument("--db-host", default="localhost",
+    help="Database hostname")
+  parser.add_argument("--db-pass", help="Database password")
+
+  return parser.parse_args()
+
+def validate_db(cursor):
```

Comment:
the one thing I'm not sure about in this PR is v`alidate_db`; I guess you are trying to move everything inside python but why not just run `mysql` as a subprocess rather than reimplement sql statement parsing? These scripts are for limited consumption so maybe this is me being picky but I suspect this is going to lead to issues down the line if there is some corner case.

I guess I would also generally be wary of including the `validate_db` call at all in a live version as it makes it easier down the line for programmer error to lead to wiping out the db. (Unless all the `DROP TABLE` stuff is removed, but it's very convenient to have it there in other workflows.) But it does seem useful for testing purposes.

Reply:
How about adding a command line option `--check-db` to activate this functionality?